src/test/java/net/openhft/chronicle/queue/impl/table/SingleTableStoreIntegrationTests.java

```diff
@@ -0,0 +1,106 @@
+package net.openhft.chronicle.queue.impl.table;
+
+import net.openhft.chronicle.core.io.IOTools;
+import net.openhft.chronicle.queue.QueueTestCommon;
+import net.openhft.chronicle.queue.impl.single.SingleChronicleQueue;
+import net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import static org.junit.Assert.assertEquals;
+
+public class SingleTableStoreIntegrationTests extends QueueTestCommon {
+
+    private TestContext context;
+
+    @Before
+    public void beforeEach() {
+        context = new TestContext();
+    }
+
+    @After
+    public void after() throws IOException {
+        context.close();
+    }
+
+    @Test
+    public void baseCasePutAndGet() {
+        context.newQueueInstance().tableStorePut("a", 1);
+        assertEquals(1, context.newQueueInstance().tableStoreGet("a"));
+    }
+
+    @Test
+    public void getMissingKeyWithoutDefault() {
+        assertEquals(Long.MIN_VALUE, context.newQueueInstance().tableStoreGet("test"));
+    }
+
+    @Test
+    public void growNumberOfKeys() {
+        SingleChronicleQueue queue1 = context.newQueueInstance();
+        queue1.tableStorePut("a", 1);
+        queue1.tableStorePut("b", 2);
+
+        SingleChronicleQueue queue2 = context.newQueueInstance();
+        queue2.tableStorePut("c", 3);
+
+        SingleChronicleQueue queue3 = context.newQueueInstance();
+        assertEquals(1, queue3.tableStoreGet("a"));
+        assertEquals(2, queue3.tableStoreGet("b"));
+        assertEquals(3, queue3.tableStoreGet("c"));
+    }
+
+    @Test
+    public void largeNumberOfKeyValuePairs() {
+        SingleChronicleQueue queue1 = context.newQueueInstance();
+        int count = 5_000;
+        for (int i = 0; i < count; i++) {
+            queue1.tableStorePut("key.prefix." + i, i);
+        }
+        for (int i = 0; i < count; i++) {
+            assertEquals(i, queue1.tableStoreGet("key.prefix." + i));
+        }
+    }
+
+    @Test
+    public void longKeyPutAndGet() {
+        SingleChronicleQueue queue1 = context.newQueueInstance();
+        StringBuilder keyBuffer = new StringBuilder();
+        Random random = new Random();
+        for (int i = 0; i < 121; i++) {
```

Comment:
Note, above this character length the table store will fail internally. A follow up PR will improve the error handling on this.